sdk/core/Azure.Core.TestFramework/src/TestEventListener.cs

```diff
@@ -14,6 +14,8 @@ public class TestEventListener : EventListener
     {
         private volatile bool _disposed;
         private readonly ConcurrentQueue<EventWrittenEventArgs> _events = new ConcurrentQueue<EventWrittenEventArgs>();
+        private uint _maxEventCount;
+        private const uint DefaultMaxEventCount = 100;
```

Comment:
This seems very low.

Reply:
1000?

Reply:
Would a level/message filter be a better solution?

Reply:
I looked at the usage of the test event listener. Seems like it is used in a very focused manner where the test code expects around a dozen events and analyzes them. We can change to 1000, though I think that's an overkill.
Level or event-id based filter would not address the root of the issue, which is the unbounded growth of this DS. For example, in the case I looked at, we were logging error level events for the one particular id we were interested in.

Reply:
I'm worried about making tests flaky and hard to debug by silently dropping messaged. I'd rather throw and have an opt in.